frontend/util.lua

```diff
@@ -400,4 +400,102 @@ function util.splitToArray(str, splitter, capture_empty_entity)
     return result
 end
 
+--- Convert a Unicode codepoint (number) to UTF8 char
+--
+--- @int c Unicode codepoint
+--- @treturn string UTF8 char
+function util.unicodeCodepointToUtf8(c)
+    if c < 128 then
+        return string.char(c)
+    elseif c < 2048 then
+        return string.char(192 + c/64, 128 + c%64)
+    elseif c < 55296 or 57343 < c and c < 65536 then
+        return string.char(224 + c/4096, 128 + c/64%64, 128 + c%64)
+    elseif c < 1114112 then
+        return string.char(240 + c/262144, 128 + c/4096%64, 128 + c/64%64, 128 + c%64)
+    else
+        return util.unicodeCodepointToUtf8(65533) -- U+FFFD REPLACEMENT CHARACTER
+    end
+end
+
+local HTML_ENTITIES_TO_UTF8 = {
+    ["&lt;"] = "<",
+    ["&gt;"] = ">",
+    ["&amp;"] = "&",
+    ["&quot;"] = '"',
+    ["&apos;"] = "'",
+    ["&nbsp;"] = "\xC2\xA0",
+    ["&#(%d+);"] = function(x) return util.unicodeCodepointToUtf8( tonumber(x) ) end,
+    ["&#x(%x+);"] = function(x) return util.unicodeCodepointToUtf8( tonumber(x,16) ) end,
+}
+--- Replace HTML entities with their UTF8 equivalent in text
+--
+-- Supports only basic ones and those with numbers (no support
+-- for named entities like &eacute;)
+--- @int string text with HTML entities
+--- @treturn string UTF8 text
+function util.htmlEntitiesToUtf8(text)
+    for k,v in pairs(HTML_ENTITIES_TO_UTF8) do
+        text = text:gsub(k, v)
+    end
+    return text
+end
+
+--- Convert HTML to plain text
```

Comment:
But I don't want to _remove Html Tags_ :)
About decode vs convert in your previous comment, I don't consider HTML as some ``encoding`` that needs to be `decoded`.
I consider it `` formatting``. And in this case, I want to convert it to another formatting (plain text), and the only HTML formatting tags that make sense in plain text are the newlines (to keep paragraphs, see the description screenshot). That's why `<br>` and `<p>` are the only ones that I handle (and if this function were needed to be super good, it should also handle ``<div>`` and ``<span style="display: block">``, and first remove HTML comments, styles and scripts, and it would be tough and long to implement that correctly).
I need it to work on simple HTML like I have found in epub's metadata ``<dc:description>``. If people were to open issues with samble books with more complex ``<dc:description>``, we may need to revisit that.